.gitignore

```diff
@@ -1,6 +1,7 @@
 # Byte-compiled / optimized / DLL files
 __pycache__/
 *.py[cod]
+*.so
```

Comment:
I'm a little worried that this needs to be added ... are you building in an unusual way or location?

Reply:
Seeing as `*.so` is also specified in [CPython's `.gitignore`](https://github.com/python/cpython/blob/7dd549eb08939e1927fba818116f5202e76f8d73/.gitignore#L9), I don't think this is an issue. 

The shared object files (`.so`) are generated when building CPython on Linux platforms in the `build/` directory, so having them in `.gitignore` ensures that they are not accidentally pushed to the repository if you build prior to opening the PR (which one should do, in order to run the regression tests locally). I don't believe it's 100% necessary (since you can just manually `git add` each changed file or directory), but it provides some convenience, such as for using `git add -A`.

Reply:
It would be wrong to include so files in the git repo, and sometimes they sneak in, so I see no reason to leave them out of the gitignore.